### PR TITLE
Fix for CASSANDRA-14218: Deprecate Throwables.propagate usage

### PR DESCRIPTION
**Pull request by Claudenw**
Replaced all occurances of `Throwables.propagate()`

### CODE REVIEW

**Comment by Claudenw** [comment]
Attached is a patch based on the fixes presented:  

[CASSANDRA-14218.txt](https://github.com/apache/cassandra/files/9349084/CASSANDRA-14218.txt)

File: src/java/org/apache/cassandra/streaming/management/ProgressInfoCompositeData.java [hunk 1]
```diff
@@ -68,7 +66,7 @@
         }
         catch (OpenDataException e)
         {
-            throw Throwables.propagate(e);
+            throw new RuntimeException(e);
```
Comment (dcapwell) [inline src/java/org/apache/cassandra/streaming/management/ProgressInfoCompositeData.java:69]: extra space

File: src/java/org/apache/cassandra/streaming/management/StreamStateCompositeData.java [hunk 2]
```diff
@@ -67,7 +66,7 @@
         }
         catch (OpenDataException e)
         {
-            throw Throwables.propagate(e);
+            throw new RuntimeException(e);
```
Comment (dcapwell) [inline src/java/org/apache/cassandra/streaming/management/StreamStateCompositeData.java:69]: extra space

File: src/java/org/apache/cassandra/tools/LoaderOptions.java [hunk 3]
```diff
@@ -174,9 +173,9 @@ public LoaderOptions build()
             }
             catch (UnknownHostException e)
             {
-                Throwables.propagate(e);
+                throw new RuntimeException(e);
             }
-
+
```
Comment (dcapwell) [inline src/java/org/apache/cassandra/tools/LoaderOptions.java:178]: extra space

**Pull request closed without merging**